### ISSUE TITLE
[DX] Add deprecated function taxonomy_get_parents_all()

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#2792]
## Describe your issue or idea
We renamed that function in Backdrop, (see change record https://api.backdropcms.org/node/41635) but it seems we missed one call in `taxonomy/views/taxonomy_views.inc`

**Comment by jlfranklin** [comment]
Sounds like we should have a deprecated wrapper function for it, if possible.

**Comment by docwilmot** [comment]
@jlfranklin all other instances of this function were simply changed in core. I suspect no one ever used the Views handler which still references this one leftover and so no one caught the error, till now. The best fix then is to just change it.

**Comment by jlfranklin** [comment]
Are there no contrib modules that used it?  Is it a private function or a public one?

**Comment by jlfranklin** [comment]
Yes, I think we should fix the last use in Views, as that's part of core and core should not call deprecated functions.  But, if it is a public function, then some Drupal 7 contrib modules probably use it and there should be a wrapper function with a `watchdog()` call logging the deprecated use.

**Comment by docwilmot** [comment]
@jlfranklin in the change record at https://api.backdropcms.org/node/41635 its noted that a bunch of similar functions were renamed before the Backdrop fork. Drupal contrib ports would of necessity need all of those changed. There are several calls to `taxonomy_term_load_parents_all()` in core which is what this function was renamed to. It wouldnt make sense to create a deprecation solution just for that one missed instance.

**Comment by jlfranklin** [comment]
If they were core Drupal 7 functions, they they all belong in the Drupal compatibility layer.  It applies equally well when renaming `taxonomy_get_term_by_name()`  to `taxonomy_term_load_multiple_by_name()` as to renaming `drupal_goto()` to `backdrop_goto()`.

**Comment by docwilmot** [comment]
I see your point. I wasnt considering the compatibility layer. Wonder why these arent included there. @quicksketch? @jenlampton?

**Comment by jenlampton** [comment]
Yes, a good point. We should add the function into the drupal compatibility layer, but with a watchdog log message, as per https://github.com/backdrop/backdrop-issues/issues/1568.

**Comment by docwilmot** [comment]
I've already pushed a PR for the Views fix.
We'd need to add all those `taxonomy_***` functions to the compatibility layer, not just that one.

**Comment by quicksketch** [comment]
I've merged the fix provided by @docwilmot (https://github.com/backdrop/backdrop/pull/1950) into 1.x and 1.7.x

> We should also add the missing function(s) into the drupal compatibility layer, but with a watchdog log message, as per #1568

The current PR actually adds the functions back to **taxonomy.module**, so they are always available even with the Drupal compatibility layer off. This matches our approach with other functions that have been deprecated during the Backdrop 1.x cycle (since they're not related to Drupal changes), but I don't know about adding "new" functions that we haven't had for the entire version back into their respective modules. It's a better place than putting it in the `drupal.inc` file though, as taxonomy.module might not even be enabled and we don't want to include every module's deprecated functions in there.

Thoughts?

**Comment by jenlampton** [comment]
Hm. I think if the "new" functions are marked as deprecated in their docblocks, it's okay to have them in `taxonomy.module`. Makes more sense than loading them when not needed, imo.

**Comment by jlfranklin** [comment]
I was trying to figure out a good place for the wrapper functions, myself.  Ideally, there would be a `taxonomy.drupal.inc` file with all the Drupal compatibility layer functions loaded when the Drupal compatibility layer is enabled, but no such thing exists and creating it would be a much more complicated patch.

In the absence of a dedicated place to put the wrappers, I defaulted to where other deprecated taxonomy functions were already.  At the very least, when we clean out all the deprecated functions in 2.x, they will all be together and we'll be less likely to miss one.

**Comment by quicksketch** [comment]
>  I defaulted to where other deprecated taxonomy functions were already.

I didn't realize we had these in there already. This makes perfect sense to me. I updated your PR slightly to reference the Backdrop change record at https://api.backdropcms.org/node/41635 instead of the Drupal.org one (though I updated our change record to cross reference the original Drupal issue). At the time, we probably didn't have the Backdrop API website yet. I also updated the `@deprecated` lines to include `1.0.0`.

I've merged https://github.com/backdrop/backdrop/pull/1954 into 1.x. This will be in Backdrop 1.8.0.
